presto-main/src/main/java/io/prestosql/connector/system/jdbc/ColumnJdbcTable.java

```diff
@@ -428,11 +426,24 @@ static Integer columnSize(Type type)
         if (type.equals(DATE)) {
             return 14; // +5881580-07-11 (2**31-1 days)
         }
-        if (type.equals(TIMESTAMP)) {
-            return 15 + 8;
-        }
-        if (type.equals(TIMESTAMP_WITH_TIME_ZONE)) {
-            return 15 + 8 + 6;
+        if (type instanceof TimestampType) {
+            // 1 digit for year sign
+            // 6 digits for year
+            // 15 digits for "-MM-DD HH:MM:SS"
+            // min(p, 1) for the fractional second period (i.e., no period if p == 0)
+            // p for the fractional digits
+            int precision = ((TimestampType) type).getPrecision();
+            return 1 + 6 + 15 + Math.min(precision, 1) + precision;
+        }
+        if (type instanceof TimestampWithTimeZoneType) {
+            // 1 digit for year sign
+            // 6 digits for year
+            // 15 digits for "-MM-DD HH:MM:SS"
+            // min(p, 1) for the fractional second period (i.e., no period if p == 0)
+            // p for the fractional digits
+            // 6 for timezone offset
```

Comment:
this can be a named zone

Reply:
I realize. Not sure what to do about it, though. Where does the current `+ 6` come from?